core/videosource.go

```diff
@@ -107,6 +107,17 @@ func (c *BasicVideoSource) UpdateHLSMasterPlaylist(manifestID ManifestID, mpl *m
 }
 
 func (c *BasicVideoSource) EvictHLSMasterPlaylist(manifestID ManifestID) {
+	c.masterPLock.Lock()
+	mpl := c.masterPList[manifestID]
+	if mpl != nil {
+		for _, variant := range mpl.Variants {
+			if len(variant.URI) > 5 {
```

Comment:
Why do we check for the URI len greater than 5?

Reply:
so next line `streamID := variant.URI[:len(variant.URI)-5]` will not throw out of range exception

Reply:
Ahh I see.  Maybe `strings.Contains()` is more obvious here?

Reply:
good point, done